hudi-common/src/main/java/org/apache/hudi/io/storage/HoodieAvroFileWriterFactory.java

```diff
@@ -66,7 +67,14 @@ protected HoodieFileWriter newParquetFileWriter(
         config.getLongOrDefault(HoodieStorageConfig.PARQUET_MAX_FILE_SIZE),
         conf.unwrapAs(Configuration.class), config.getDoubleOrDefault(HoodieStorageConfig.PARQUET_COMPRESSION_RATIO_FRACTION),
         config.getBooleanOrDefault(HoodieStorageConfig.PARQUET_DICTIONARY_ENABLED));
-    return new HoodieAvroParquetWriter(path, parquetConfig, instantTime, taskContextSupplier, populateMetaFields);
+    try {
+      return (HoodieFileWriter) ReflectionUtils.loadClass("org.apache.hudi.io.storage.HoodieAvroParquetWriter",
```

Comment:
Similarly, define such implementation class names, e.g., `"org.apache.hudi.io.storage.HoodieAvroParquetWriter"`, in a common place, so it's easier to replace later with configs.